src/main/java/duckling/Server.java

```diff
@@ -1,11 +1,101 @@
 package duckling;
 
+import duckling.requests.Request;
+import duckling.requests.RequestBuilder;
+import duckling.responders.Responders;
+
+import java.io.*;
+import java.net.*;
+import java.util.concurrent.*;
+
 public class Server {
+    public static final String CLRF = "\r\n";
+
     public int port;
+    public String root;
+
+    private boolean shuttingDown = false;
+    private InetSocketAddress address;
+    private ServerSocket connection;
+    private ExecutorService pool;
+
+    public Server(int port, String root) throws IOException {
+        this(
+            port,
+            root,
+            new ServerSocket(),
+            Executors.newCachedThreadPool()
+        );
+    }
 
-    public Server(int port) {
+    public Server(
+        int port,
+        String root,
+        ServerSocket connection,
+        ExecutorService pool
+    ) throws IOException {
         this.port = port;
+        this.root = root;
+
+        this.address = new InetSocketAddress(this.port);
+        this.connection = connection;
+        this.pool = pool;
+    }
+
+    public boolean isPoolClosed() {
+        return this.pool.isShutdown();
+    }
+
+    public void onBegin() throws IOException {
+        this.connection.bind(this.address);
+    }
+
+    public void onRequest() throws IOException {
+        Socket client = this.connection.accept();
+        Runnable handler = () -> {
+            try {
+                RequestBuilder builder = new RequestBuilder(
+                    this.root,
+                    client.getInputStream()
+                );
+                Request request = builder.build();
+                Responders responders = new Responders(
+                    request,
+                    client.getOutputStream()
+                );
+
+                builder.printRawRequest();
+                responders.getResponder().respond();
+
+                client.close();
+            } catch (IOException exception) {
+                exception.printStackTrace();
+            }
+        };
+
+        this.pool.execute(handler);
+    }
```

Comment:
This method is getting pretty large - I wonder whether there are methods you could extract or other ways to simplify. Great job limiting column width though!